### DISCUSSION TITLE
Validation on Change Based on Another Field

### DISCUSSION THREAD

**Discussion by yousinix** [mantinedev/mantine#4770] [Feature requests]
### What package has an issue

@mantine/form

### Describe the bug

This example is the same as the [Validation based on other form values](https://mantine.dev/form/validation/#validation-based-on-other-form-values) example but with `validateInputOnChange` set to `true`.

### Reproduction steps

1. Enter `mant` in the Password field.
1. Enter `mantine` in the Confirm Password field: Error shows up
1. Edit the Password field to `mantine`: Error should disappear from Confirm Password

### What version of @mantine/hooks page do you have in package.json?

5.5.6

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/sleepy-carlos-y8fixh?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Maybe (with some guidance)

**Comment by rtivital** [comment]
`validateInputOnChange` validates only field that you are currently changing. For more complex logic, you should use `form.validateField` – https://codesandbox.io/s/romantic-proskuriakova-ctgu1t?file=/src/App.tsx

**Comment by yousinix** [comment]
Thanks for the suggestion!

I just feel this behavior isweird, comparing it to how formik handles things:
https://codesandbox.io/s/l2r832l8x7
